src/cli/commands/install.js

```diff
@@ -483,13 +483,6 @@ export class Install {
         break;
       }
     }
-    // check if loaded lockfile has patterns we don't have, eg. uninstall
```

Comment:
patterns contain only direct dependencies.
Lockfile contains all dependencies, it will always have more than the ones from package.json.